src/Output/ProgressBar.php

```diff
@@ -0,0 +1,300 @@
+     */
+    protected int $total = 0;
+
+    /**
+     * The current item that the progress bar represents
+     */
+    protected int $current = 0;
+
+    /**
+     * The current percentage displayed
+     */
+    protected string $currentPercentage = '';
+
+    /**
+     * The string length of the bar when at 100%
+     */
+    protected int $barStrLen = 0;
+
+    /**
+     * Flag indicating whether we are writing the bar for the first time
+     */
+    protected bool $firstLine = true;
+
+    /**
+     * Current status bar label
+     */
+    protected string $label = '';
+
+    /**
+     * Options for progress bar
+     *
+     * @todo define setter that user can override for custom progress bar
+     */
+    private array $options = [
+        'pointer'    => '>',
+        'loader'     => '=',
+        'labelColor' => 'white',
+    ];
+
+    /**
+     * Force a redraw every time
+     */
+    protected bool $forceRedraw = false;
+
+    /**
+     * If this progress bar ever displayed a label.
+     */
+    protected bool $hasLabelLine = false;
+
+    protected Writer $writer;
+
+    protected Cursor $cursor;
+
+    protected Terminal $terminal;
+
+    /**
+     * If they pass in a total, set the total
+     */
+    public function __construct(?int $total = null, ?Writer $writer = null)
+    {
+        if ($total !== null) {
+            $this->total($total);
+        }
+
+        $this->writer   = $writer ?: new Writer();
+        $this->cursor   = $this->writer->cursor();
+        $this->terminal = new Terminal();
+    }
+
+    /**
+     * Set the total property
+     */
+    public function total(int $total): self
+    {
+        $this->total = $total;
+
+        return $this;
+    }
+
+    /**
+     * Determines the current percentage we are at and re-writes the progress bar
+     *
+     * @return void
+     * @throws UnexpectedValueException
+     */
+    public function current(int $current, string $label = '')
+    {
+        if ($this->total == 0) {
+            // Avoid dividing by 0
+            throw new UnexpectedValueException('The progress total must be greater than zero.');
+        }
+
+        if ($current > $this->total) {
+            throw new UnexpectedValueException('The current is greater than the total.');
+        }
+
+        $this->drawProgressBar($current, $label);
+
+        $this->current = $current;
+        $this->label   = $label;
+    }
+
+    /**
+     * Increments the current position we are at and re-writes the progress bar
+     *
+     * @param integer $increment The number of items to increment by
+     */
+    public function advance(int $increment = 1, string $label = '')
+    {
+        $this->current($this->current + $increment, $label);
+    }
+
+    /**
+     * Force the progress bar to redraw every time regardless of whether it has changed or not
+     */
+    public function forceRedraw(bool $force = true): self
+    {
+        $this->forceRedraw = !!$force;
```

Comment:
```suggestion
        $this->forceRedraw = $force;
```

Reply:
well seen. I don't even know why I wrote it like that 😂